math/least_common_multiple.cpp

```diff
@@ -0,0 +1,41 @@
+// Copyright 2020 @author tjgurwara99
+
+#include <iostream>
+
+/*
+ * First we create a function for greatest common divisor,
+ * there are many ways of implementing this function. I chose to
+ * use recursive function to reduce unnecessary loops.
+ */
+unsigned int gcd(unsigned int x, unsigned int y) {
+    if (x == 0) {
+        return y;
+    }
+    if (y == 0) {
+        return x;
+    }
+    if (x == y) {
+        return x;
+    }
+    if (x > y) {
+        // The following is valid because we have checked whether y == 0
+
+        int temp = x / y;
+        return gcd(y, x - temp * y);
+    }
+    // Again the following is valid because we have checked whether x == 0
+
+    int temp = y / x;
+    return gcd(x, y - temp * x);
+}
+
+/*
+ * Now we use the fact that lcm(x,y) * gcd(x,y)  = x * y
+ */
+unsigned int lcm(unsigned int x, unsigned int y) { return x * y / gcd(x, y); }
+
+int main() {
+    int ans = lcm(5, 10);
```

Comment:
i suggest you to write some test cases to validate the algorithm properly, please add a function `void tests()` and write some tests in this functions using `assert` statements and call this function from `main()`